src/node/kernel_notifications.h

```diff
@@ -59,12 +59,14 @@ class KernelNotifications : public kernel::Notifications
     //! The block for which the last blockTip notification was received.
     //! It's first set when the tip is connected during node initialization.
     //! Might be unset during an early shutdown.
-    uint256 m_tip_block GUARDED_BY(m_tip_block_mutex){uint256::ZERO};
+    std::optional<uint256> TipBlock() EXCLUSIVE_LOCKS_REQUIRED(m_tip_block_mutex);
 
 private:
     const std::function<bool()>& m_shutdown_request;
     std::atomic<int>& m_exit_status;
     node::Warnings& m_warnings;
+
+    std::optional<uint256> m_tip_block GUARDED_BY(m_tip_block_mutex){};
```

Comment:
I could be mistaken, but the `{}` initializer here might be initializing to uint256::ZERO instead of `nullopt`. Intended?

Reply:
I'll drop the `{}` so it should be clear that it inits to `nullopt`. I'm fairly sure the current syntax also does that and is not the same as `{uint256{}}`.